cmd/sifnoded/main.go

```diff
@@ -15,31 +18,109 @@ import (
 
 	"github.com/Sifchain/sifnode/app"
 
+	"github.com/Sifchain/sifnode/cmd/ebrelayer/siflogger"
 	"github.com/cosmos/cosmos-sdk/baseapp"
 	"github.com/cosmos/cosmos-sdk/client/debug"
 	"github.com/cosmos/cosmos-sdk/client/flags"
 	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/server/config"
 	"github.com/cosmos/cosmos-sdk/store"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
 	"github.com/cosmos/cosmos-sdk/x/auth"
 	genutilcli "github.com/cosmos/cosmos-sdk/x/genutil/client/cli"
 	"github.com/cosmos/cosmos-sdk/x/staking"
+	tcmd "github.com/tendermint/tendermint/cmd/tendermint/commands"
+
+	cfg "github.com/tendermint/tendermint/config"
 )
 
 const flagInvCheckPeriod = "inv-check-period"
 
 var invCheckPeriod uint
 
+// If a new config is created, change some of the default tendermint settings
+func interceptLoadConfig() (conf *cfg.Config, err error) {
+	tmpConf := cfg.DefaultConfig()
+	err = viper.Unmarshal(tmpConf)
+	if err != nil {
+		// TODO: Handle with #870
+		panic(err)
+	}
+	rootDir := tmpConf.RootDir
+	configFilePath := filepath.Join(rootDir, "config/config.toml")
+	// Intercept only if the file doesn't already exist
+
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		// the following parse config is needed to create directories
+		conf, _ = tcmd.ParseConfig() // NOTE: ParseConfig() creates dir/files as necessary.
+		conf.ProfListenAddress = "localhost:6060"
+		conf.P2P.RecvRate = 5120000
+		conf.P2P.SendRate = 5120000
+		conf.TxIndex.IndexAllKeys = true
+		conf.Consensus.TimeoutCommit = 5 * time.Second
+		cfg.WriteConfigFile(configFilePath, conf)
+		// Fall through, just so that its parsed into memory.
+	}
+
+	if conf == nil {
+		conf, err = tcmd.ParseConfig() // NOTE: ParseConfig() creates dir/files as necessary.
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	appConfigFilePath := filepath.Join(rootDir, "config/app.toml")
+	if _, err := os.Stat(appConfigFilePath); os.IsNotExist(err) {
+		appConf, _ := config.ParseConfig()
+		config.WriteConfigFile(appConfigFilePath, appConf)
+	}
+
+	viper.SetConfigName("app")
+	err = viper.MergeInConfig()
+
+	return conf, err
+}
+func persistentPreRunE(context *server.Context) func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if cmd.Name() == version.Cmd.Name() {
+			return nil
+		}
+		// set up config
+		config, err := interceptLoadConfig()
+		if err != nil {
+			return err
+		}
+		context.Config = config
+
+		// set up logger
+		logger := siflogger.New(siflogger.TDFmt)
+		if viper.GetBool(cli.TraceFlag) {
+			// TODO: to implement stack tracing using NewTracingLogger or its parts
+			panic("TraceStack is not implemented")
+		}
+
+		err = logger.SetFilterForLayerFromConfig("module", config.LogLevel, cfg.DefaultLogLevel())
+		if err != nil {
+			return err
+		}
+		context.Logger = logger.With("module", "main")
+
+		return nil
+	}
+}
+
```

Comment:
Probably shoulld add this to simapp as well

Reply:
did I understand correctly that simapp is used only for testing purposes? Since I have not found any entry point except test_ functions.
And just in case to clarify again: this logger will be used **one per process**, not per module and not per package. As I understand by now we have only 2 processes overall to run chain: ebrelayer and sifnoded.
Is that correct?

Regarding simapp I can add it as well but it does not give any value if simapp is used only for dry testing (or please clarify if I got it wrong?).

Reply:
The simapp can be run seprately
You are right it is just for testing , but it might still be usefull to be able redirect output with loglevels instead.
I do realise it does not need to be part of this pr , we can add that in seperatly,